plugins/inputs/logfile/tailersrc.go

```diff
@@ -273,24 +265,88 @@ func (ts *tailerSrc) runTail() {
 				cnt++
 			}
 
-			if cnt < 5 {
-				continue
+			if cnt >= 5 {
+				ts.publishEvent(msgBuf, fo)
+				msgBuf.Reset()
+				cnt = 0
 			}
+		case <-ts.done:
+			return
+		}
+	}
+}
+
+func (ts *tailerSrc) publishEvent(msgBuf bytes.Buffer, fo *fileOffset) {
+	// helper to handle event publishing
+	if msgBuf.Len() == 0 {
+		return
+	}
+	msg := msgBuf.String()
+	timestamp, modifiedMsg := ts.timestampFn(msg)
+	e := &LogEvent{
+		msg:    modifiedMsg,
+		t:      timestamp,
+		offset: *fo,
+		src:    ts,
+	}
+	if ShouldPublish(ts.group, ts.stream, ts.filters, e) {
+		if ts.backpressureFdDrop {
+			select {
+			case ts.buffer <- e:
+				// successfully sent
+			case <-ts.done:
+				return
+			default:
+				// sender buffer is full. start timer to close file then retry
+				timer := time.NewTimer(tailCloseThreshold)
+				defer timer.Stop()
+
+				for {
+					select {
+					case ts.buffer <- e:
+						// sent event after buffer gets freed up
+						if ts.tailer.IsFileClosed() { // skip file closing if not already closed
+							if err := ts.tailer.Reopen(false); err != nil {
+								log.Printf("E! [logfile] error reopening file %s: %v", ts.tailer.Filename, err)
```

Comment:
nit: We might not want to log this if it just wasn't able to find the file anymore. Might be misleading.